lang/c++/impl/DataFile.cc

```diff
@@ -243,13 +243,13 @@ void DataFileWriterBase::flush()
     sync();
 }
 
-boost::mt19937 random(static_cast<uint32_t>(time(0)));
+boost::mt19937 random(static_cast<uint32_t>(time(nullptr)));
 
 DataFileSync DataFileWriterBase::makeSync()
 {
     DataFileSync sync;
-    for (size_t i = 0; i < sync.size(); ++i) {
-        sync[i] = random();
+    for (unsigned char & i : sync) {
```

Comment:
this is less straightforward to me than the previous code.  If you want to refactor, I think `std::generate` would work here and be clearer.

Reply:
Nice. Thanks, done.

Reply:
Nice. Thanks, done.